File: ds-algo-js/1.js
//  Reverse the array
//  First approch
const ReverseArray = (arr) => {
  var temp;

  var arrSize = arr.length - 1;

  for (let i = 0; i <= arrSize; arrSize--, i++) {
    temp = arr[arrSize];
    arr[arrSize] = arr[i];
    arr[i] = temp;
  }

  console.log(arr);
};

ReverseArray([1, 2, 3, 4, 5]);

//  Second Approch
var myArray = [1, 2, 3, 4, 5];
console.log("Original Array is : ");
console.log(myArray);
var revArray = myArray.reverse();
console.log("Reversed Array is : ");
console.log(revArray);

File: ds-algo-js/1.cpp
// Reverse The array
// First Approch
//  
#include <iostream>
using namespace std;
// Function for printing the array
void PrintArray(int arr[], int arrSize)
{
    cout << "{ ";
    for (int i = 0; i < arrSize; i++)
    {
        cout << arr[i] << ", ";
    }
    cout << "}" << endl;
}

// Function declaration for reversing the array
void ReverseArray(int arr[], int arrSize)
{
    int temp;
    arrSize--;
    for (int i = 0; i <= arrSize; arrSize--, i++)
    {
        temp = arr[i];
        arr[i] = arr[arrSize];
        arr[arrSize] = temp;
    }
}

int main()
{
    int arr[] = {1, 2, 3, 4, 5};
    int arrSize = sizeof(arr) / sizeof(arr[0]);

    cout << "The original Array is : " << endl;
    // Printing original array
    PrintArray(arr, arrSize);
    // Reverse the array
    ReverseArray(arr, arrSize);
    cout << "The reverse Array is : " << endl;
    // Printing reverse array
    PrintArray(arr, arrSize);

    return 0;
}